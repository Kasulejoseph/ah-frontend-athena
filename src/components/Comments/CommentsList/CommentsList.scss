.reply-list {
  display: block !important;
}
.mdc-chip {
  margin-left: 60%;
  padding-right: 15%;
  color: #6c757d;
  max-width: max-content;
}
.square {
  width: 50px !important;
  height: 50px !important;
}
.rounded-circle {
  border-radius: 50% !important;
}
.comment-body{
  margin-bottom: 5%;
}
.user-info  {
  padding-left: 0px !important;
  padding-right: 0px;
  background-color: #f5f5f5;
}
.mdc-chip {
  margin-left: 5px !important;
  padding: 0 5px 5px 5px !important;
}
.date {
  margin-left: 10px !important;
}

