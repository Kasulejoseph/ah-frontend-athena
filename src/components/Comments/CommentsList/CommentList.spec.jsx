import React from 'react';
import { shallow } from 'enzyme';
import { CommentsList, mapStateToProps } from './CommentsList';
import LoadReply from './LoadReply';

let props = {
  commentList: [
    {
      replies: {},
      author: { image: '' }
    },
    {
      replies: {},
      author: { image: '' }
    }
  ],
  showReplies: 'd-none',
  showComBox: 'd-none',
  isEdit: false,
  CommentDeleteAction: () => {},
  dateToHours: () => {}
};
const commentData = {
  comment: {
    commentList: [
      {
        comment_body: 'my comment',
        author: { image: 'url' }
      }
    ]
  }
};

describe('CommentsList test', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<CommentsList {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should handle clickReply function sets showcomBox to d-none', () => {
    const wrapper = shallow(<CommentsList {...props} />);
    wrapper.setState({ showComBox: '' });
    wrapper.instance().clickReply();
    expect(wrapper.state('showComBox')).toEqual('d-none');
  });
  it('should handle clickReply function sets showcomBox to none', () => {
    const wrapper = shallow(<CommentsList {...props} />);
    wrapper.instance().clickReply();
    expect(wrapper.state('showComBox')).toEqual('');
  });

  it('should handle clickDelete', () => {
    const wrapper = shallow(<CommentsList {...props} />);
    wrapper.instance().clickDelete(1);
    expect(wrapper.state('showReplies')).toEqual('d-none');
  });

  it('should handle clickEdit and set edit to true', () => {
    const wrapper = shallow(<CommentsList {...props} />);
    wrapper.setState({ showComBox: '' });
    wrapper.instance().clickEdit(1);
    expect(wrapper.state('showComBox')).toEqual('d-none');
  });

  it('should handle clickEdit when showcomBox is set to d-none', () => {
    const wrapper = shallow(<CommentsList {...props} />);
    wrapper.setState({ showReplies: '' });
    wrapper.instance().clickEdit(1);
    expect(wrapper.state('showReplies')).toEqual('');
  });

  it('should handle repliesShow and set repliesShow to d-none', () => {
    const wrapper = shallow(<CommentsList {...props} />);
    wrapper.setState({ showReplies: '' });
    wrapper.instance().repliesShow();
    expect(wrapper.state('showReplies')).toEqual('d-noneundefined');
  });

  it('should handle repliesShow and set repliesShow to none', () => {
    const wrapper = shallow(<CommentsList {...props} />);
    wrapper.setState({ showReplies: 'd-none' });
    wrapper.instance().repliesShow();
    expect(wrapper.state('showReplies')).toEqual('');
  });

  it('should handle dateToHours', () => {
    const wrapper = shallow(<CommentsList {...props} />);
    wrapper.instance().dateToHours();
    expect(wrapper.state('isEdit')).toEqual(false);
  });

  it('should handle onchange function', () => {
    const wrapper = shallow(<CommentsList {...props} />);
    wrapper
      .instance()
      .onChange({ target: { name: 'comment', value: 'comment_body' } });
    expect(wrapper.state('comment')).toEqual('comment_body');
  });

  it('should will recieve comments props', () => {
    const wrapper = shallow(<CommentsList {...props} />);
    wrapper.setProps({ commentData });
    expect(wrapper.state()).toEqual({
      commentList: [
        { author: { image: '' }, replies: {} },
        { author: { image: '' }, replies: {} }
      ],
      isEdit: false,
      modal: false,
      showComBox: 'd-none',
      showReplies: 'd-none'
    });
  });

  it('should will recieve comments props with a refresh', () => {
    const refresh = true;
    const wrapper = shallow(<CommentsList {...props} refresh={refresh} />);
    wrapper.setProps({ refresh: true });
    wrapper.setProps({ commentData });
    expect(wrapper.state('commentList')).toEqual([
      { author: { image: '' }, replies: {} },
      { author: { image: '' }, replies: {} }
    ]);
  });

  it('should map state to props', () => {
    expect(mapStateToProps(commentData).commentList).toEqual([
      { author: { image: 'url' }, comment_body: 'my comment' }
    ]);
  });
});

describe('test load reply', () => {
  it('should take a snapshot', () => {
    const wrapper = shallow(<LoadReply {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
