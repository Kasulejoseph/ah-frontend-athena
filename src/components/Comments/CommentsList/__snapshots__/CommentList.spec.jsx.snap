// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentsList test matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommentsList
    CommentDeleteAction={[Function]}
    CommentGetAction={[Function]}
    commentList={
      Array [
        Object {
          "author": Object {
            "image": "",
          },
          "replies": Object {},
        },
        Object {
          "author": Object {
            "image": "",
          },
          "replies": Object {},
        },
      ]
    }
    data={Object {}}
    dateToHours={[Function]}
    isEdit={false}
    refresh={false}
    showComBox="d-none"
    showReplies="d-none"
    slug={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(ReplyBox)
          art_slug={Object {}}
          childId={null}
          isEdit={false}
          parentId={null}
        />,
        <ModalPage
          fallback="/"
          md="12"
          modal={false}
          title="Please login before you can rate the article"
          toggle={[Function]}
        />,
        Array [],
      ],
      "className": "m-1 p-1 container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "art_slug": Object {},
          "childId": null,
          "isEdit": false,
          "parentId": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "fallback": "/",
          "history": undefined,
          "md": "12",
          "modal": false,
          "title": "Please login before you can rate the article",
          "toggle": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(ReplyBox)
            art_slug={Object {}}
            childId={null}
            isEdit={false}
            parentId={null}
          />,
          <ModalPage
            fallback="/"
            md="12"
            modal={false}
            title="Please login before you can rate the article"
            toggle={[Function]}
          />,
          Array [],
        ],
        "className": "m-1 p-1 container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "art_slug": Object {},
            "childId": null,
            "isEdit": false,
            "parentId": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "fallback": "/",
            "history": undefined,
            "md": "12",
            "modal": false,
            "title": "Please login before you can rate the article",
            "toggle": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`test load reply should take a snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoadReply
    CommentDeleteAction={[Function]}
    art_slug="mock-slug1"
    childId={1}
    comment={Object {}}
    commentList={
      Array [
        Object {
          "author": Object {
            "image": "",
          },
          "replies": Object {},
        },
        Object {
          "author": Object {
            "image": "",
          },
          "replies": Object {},
        },
      ]
    }
    dateToHours={[Function]}
    img="url"
    isEdit={false}
    onKeyUp={[Function]}
    parentId={1}
    parentId2={1}
    replyList={Array []}
    showComBox="d-none"
    showReplies="d-none"
    username="kasule"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <t
        className="mt-3 p-0 m-0"
        fluid={false}
        tag="div"
      >
        <t
          className="mw-80"
          tag="div"
        >
          <t
            className="mw-100"
            tag="div"
          >
            <h6
              className="comment-body"
            />
            <div
              className="col-3 user-info  offset-9 border border-white"
            >
              <div
                className="date"
              >
                <t
                  className="mr-2"
                  color="light"
                  pill={false}
                  tag="span"
                />
                Hours ago
              </div>
              <div
                className="mdc-chip"
              >
                <img
                  alt="Contact Person"
                  className="img-fluid z-depth-1 square mr-2  rounded-circle"
                  src="url"
                />
                kasule
              </div>
            </div>
            <CommentEdit />
            <div
              className="col-4 text-primary"
              onKeyUp={[Function]}
              role="button"
              tabIndex="0"
            >
              view all...
            </div>
            <div
              className="d-none"
            >
              <Connect(ReplyList)
                CommentDeleteAction={[Function]}
                parentId={1}
                replyList={Array []}
                slug="mock-slug1"
              />
            </div>
            <div
              className="d-none"
            >
              <Connect(ReplyBox)
                art_slug="mock-slug1"
                childId={1}
                isEdit={false}
                parentId={1}
              />
            </div>
          </t>
        </t>
      </t>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <t
          className="mw-80"
          tag="div"
        >
          <t
            className="mw-100"
            tag="div"
          >
            <h6
              className="comment-body"
            />
            <div
              className="col-3 user-info  offset-9 border border-white"
            >
              <div
                className="date"
              >
                <t
                  className="mr-2"
                  color="light"
                  pill={false}
                  tag="span"
                />
                Hours ago
              </div>
              <div
                className="mdc-chip"
              >
                <img
                  alt="Contact Person"
                  className="img-fluid z-depth-1 square mr-2  rounded-circle"
                  src="url"
                />
                kasule
              </div>
            </div>
            <CommentEdit />
            <div
              className="col-4 text-primary"
              onKeyUp={[Function]}
              role="button"
              tabIndex="0"
            >
              view all...
            </div>
            <div
              className="d-none"
            >
              <Connect(ReplyList)
                CommentDeleteAction={[Function]}
                parentId={1}
                replyList={Array []}
                slug="mock-slug1"
              />
            </div>
            <div
              className="d-none"
            >
              <Connect(ReplyBox)
                art_slug="mock-slug1"
                childId={1}
                isEdit={false}
                parentId={1}
              />
            </div>
          </t>
        </t>,
        "className": "mt-3 p-0 m-0",
        "fluid": false,
        "tag": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <t
            className="mw-100"
            tag="div"
          >
            <h6
              className="comment-body"
            />
            <div
              className="col-3 user-info  offset-9 border border-white"
            >
              <div
                className="date"
              >
                <t
                  className="mr-2"
                  color="light"
                  pill={false}
                  tag="span"
                />
                Hours ago
              </div>
              <div
                className="mdc-chip"
              >
                <img
                  alt="Contact Person"
                  className="img-fluid z-depth-1 square mr-2  rounded-circle"
                  src="url"
                />
                kasule
              </div>
            </div>
            <CommentEdit />
            <div
              className="col-4 text-primary"
              onKeyUp={[Function]}
              role="button"
              tabIndex="0"
            >
              view all...
            </div>
            <div
              className="d-none"
            >
              <Connect(ReplyList)
                CommentDeleteAction={[Function]}
                parentId={1}
                replyList={Array []}
                slug="mock-slug1"
              />
            </div>
            <div
              className="d-none"
            >
              <Connect(ReplyBox)
                art_slug="mock-slug1"
                childId={1}
                isEdit={false}
                parentId={1}
              />
            </div>
          </t>,
          "className": "mw-80",
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <h6
                className="comment-body"
              />,
              <div
                className="col-3 user-info  offset-9 border border-white"
              >
                <div
                  className="date"
                >
                  <t
                    className="mr-2"
                    color="light"
                    pill={false}
                    tag="span"
                  />
                  Hours ago
                </div>
                <div
                  className="mdc-chip"
                >
                  <img
                    alt="Contact Person"
                    className="img-fluid z-depth-1 square mr-2  rounded-circle"
                    src="url"
                  />
                  kasule
                </div>
              </div>,
              <CommentEdit />,
              <div
                className="col-4 text-primary"
                onKeyUp={[Function]}
                role="button"
                tabIndex="0"
              >
                view all...
              </div>,
              <div
                className="d-none"
              >
                <Connect(ReplyList)
                  CommentDeleteAction={[Function]}
                  parentId={1}
                  replyList={Array []}
                  slug="mock-slug1"
                />
              </div>,
              <div
                className="d-none"
              >
                <Connect(ReplyBox)
                  art_slug="mock-slug1"
                  childId={1}
                  isEdit={false}
                  parentId={1}
                />
              </div>,
            ],
            "className": "mw-100",
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "comment-body",
              },
              "ref": null,
              "rendered": null,
              "type": "h6",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="date"
                  >
                    <t
                      className="mr-2"
                      color="light"
                      pill={false}
                      tag="span"
                    />
                    Hours ago
                  </div>,
                  <div
                    className="mdc-chip"
                  >
                    <img
                      alt="Contact Person"
                      className="img-fluid z-depth-1 square mr-2  rounded-circle"
                      src="url"
                    />
                    kasule
                  </div>,
                ],
                "className": "col-3 user-info  offset-9 border border-white",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <t
                        className="mr-2"
                        color="light"
                        pill={false}
                        tag="span"
                      />,
                      "Hours ago",
                    ],
                    "className": "date",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": undefined,
                        "className": "mr-2",
                        "color": "light",
                        "pill": false,
                        "tag": "span",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "Hours ago",
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <img
                        alt="Contact Person"
                        className="img-fluid z-depth-1 square mr-2  rounded-circle"
                        src="url"
                      />,
                      "kasule",
                    ],
                    "className": "mdc-chip",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "alt": "Contact Person",
                        "className": "img-fluid z-depth-1 square mr-2  rounded-circle",
                        "src": "url",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                    "kasule",
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "CommentEditAction": undefined,
                "clickDelete": undefined,
                "clickDisLike": undefined,
                "clickEdit": undefined,
                "clickLike": undefined,
                "clickReply": undefined,
                "id": undefined,
                "likes": undefined,
                "onClick": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "view all...",
                "className": "col-4 text-primary",
                "onClick": undefined,
                "onKeyUp": [Function],
                "role": "button",
                "tabIndex": "0",
              },
              "ref": null,
              "rendered": "view all...",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(ReplyList)
                  CommentDeleteAction={[Function]}
                  parentId={1}
                  replyList={Array []}
                  slug="mock-slug1"
                />,
                "className": "d-none",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "CommentDeleteAction": [Function],
                  "history": undefined,
                  "onSubmitReply": undefined,
                  "parentId": 1,
                  "repliesShow": undefined,
                  "replyList": Array [],
                  "slug": "mock-slug1",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(ReplyBox)
                  art_slug="mock-slug1"
                  childId={1}
                  isEdit={false}
                  parentId={1}
                />,
                "className": "d-none",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "art_slug": "mock-slug1",
                  "childId": 1,
                  "isEdit": false,
                  "parentId": 1,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <t
          className="mt-3 p-0 m-0"
          fluid={false}
          tag="div"
        >
          <t
            className="mw-80"
            tag="div"
          >
            <t
              className="mw-100"
              tag="div"
            >
              <h6
                className="comment-body"
              />
              <div
                className="col-3 user-info  offset-9 border border-white"
              >
                <div
                  className="date"
                >
                  <t
                    className="mr-2"
                    color="light"
                    pill={false}
                    tag="span"
                  />
                  Hours ago
                </div>
                <div
                  className="mdc-chip"
                >
                  <img
                    alt="Contact Person"
                    className="img-fluid z-depth-1 square mr-2  rounded-circle"
                    src="url"
                  />
                  kasule
                </div>
              </div>
              <CommentEdit />
              <div
                className="col-4 text-primary"
                onKeyUp={[Function]}
                role="button"
                tabIndex="0"
              >
                view all...
              </div>
              <div
                className="d-none"
              >
                <Connect(ReplyList)
                  CommentDeleteAction={[Function]}
                  parentId={1}
                  replyList={Array []}
                  slug="mock-slug1"
                />
              </div>
              <div
                className="d-none"
              >
                <Connect(ReplyBox)
                  art_slug="mock-slug1"
                  childId={1}
                  isEdit={false}
                  parentId={1}
                />
              </div>
            </t>
          </t>
        </t>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <t
            className="mw-80"
            tag="div"
          >
            <t
              className="mw-100"
              tag="div"
            >
              <h6
                className="comment-body"
              />
              <div
                className="col-3 user-info  offset-9 border border-white"
              >
                <div
                  className="date"
                >
                  <t
                    className="mr-2"
                    color="light"
                    pill={false}
                    tag="span"
                  />
                  Hours ago
                </div>
                <div
                  className="mdc-chip"
                >
                  <img
                    alt="Contact Person"
                    className="img-fluid z-depth-1 square mr-2  rounded-circle"
                    src="url"
                  />
                  kasule
                </div>
              </div>
              <CommentEdit />
              <div
                className="col-4 text-primary"
                onKeyUp={[Function]}
                role="button"
                tabIndex="0"
              >
                view all...
              </div>
              <div
                className="d-none"
              >
                <Connect(ReplyList)
                  CommentDeleteAction={[Function]}
                  parentId={1}
                  replyList={Array []}
                  slug="mock-slug1"
                />
              </div>
              <div
                className="d-none"
              >
                <Connect(ReplyBox)
                  art_slug="mock-slug1"
                  childId={1}
                  isEdit={false}
                  parentId={1}
                />
              </div>
            </t>
          </t>,
          "className": "mt-3 p-0 m-0",
          "fluid": false,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <t
              className="mw-100"
              tag="div"
            >
              <h6
                className="comment-body"
              />
              <div
                className="col-3 user-info  offset-9 border border-white"
              >
                <div
                  className="date"
                >
                  <t
                    className="mr-2"
                    color="light"
                    pill={false}
                    tag="span"
                  />
                  Hours ago
                </div>
                <div
                  className="mdc-chip"
                >
                  <img
                    alt="Contact Person"
                    className="img-fluid z-depth-1 square mr-2  rounded-circle"
                    src="url"
                  />
                  kasule
                </div>
              </div>
              <CommentEdit />
              <div
                className="col-4 text-primary"
                onKeyUp={[Function]}
                role="button"
                tabIndex="0"
              >
                view all...
              </div>
              <div
                className="d-none"
              >
                <Connect(ReplyList)
                  CommentDeleteAction={[Function]}
                  parentId={1}
                  replyList={Array []}
                  slug="mock-slug1"
                />
              </div>
              <div
                className="d-none"
              >
                <Connect(ReplyBox)
                  art_slug="mock-slug1"
                  childId={1}
                  isEdit={false}
                  parentId={1}
                />
              </div>
            </t>,
            "className": "mw-80",
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <h6
                  className="comment-body"
                />,
                <div
                  className="col-3 user-info  offset-9 border border-white"
                >
                  <div
                    className="date"
                  >
                    <t
                      className="mr-2"
                      color="light"
                      pill={false}
                      tag="span"
                    />
                    Hours ago
                  </div>
                  <div
                    className="mdc-chip"
                  >
                    <img
                      alt="Contact Person"
                      className="img-fluid z-depth-1 square mr-2  rounded-circle"
                      src="url"
                    />
                    kasule
                  </div>
                </div>,
                <CommentEdit />,
                <div
                  className="col-4 text-primary"
                  onKeyUp={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  view all...
                </div>,
                <div
                  className="d-none"
                >
                  <Connect(ReplyList)
                    CommentDeleteAction={[Function]}
                    parentId={1}
                    replyList={Array []}
                    slug="mock-slug1"
                  />
                </div>,
                <div
                  className="d-none"
                >
                  <Connect(ReplyBox)
                    art_slug="mock-slug1"
                    childId={1}
                    isEdit={false}
                    parentId={1}
                  />
                </div>,
              ],
              "className": "mw-100",
              "tag": "div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "comment-body",
                },
                "ref": null,
                "rendered": null,
                "type": "h6",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="date"
                    >
                      <t
                        className="mr-2"
                        color="light"
                        pill={false}
                        tag="span"
                      />
                      Hours ago
                    </div>,
                    <div
                      className="mdc-chip"
                    >
                      <img
                        alt="Contact Person"
                        className="img-fluid z-depth-1 square mr-2  rounded-circle"
                        src="url"
                      />
                      kasule
                    </div>,
                  ],
                  "className": "col-3 user-info  offset-9 border border-white",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <t
                          className="mr-2"
                          color="light"
                          pill={false}
                          tag="span"
                        />,
                        "Hours ago",
                      ],
                      "className": "date",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": undefined,
                          "className": "mr-2",
                          "color": "light",
                          "pill": false,
                          "tag": "span",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      "Hours ago",
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <img
                          alt="Contact Person"
                          className="img-fluid z-depth-1 square mr-2  rounded-circle"
                          src="url"
                        />,
                        "kasule",
                      ],
                      "className": "mdc-chip",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "alt": "Contact Person",
                          "className": "img-fluid z-depth-1 square mr-2  rounded-circle",
                          "src": "url",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "img",
                      },
                      "kasule",
                    ],
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "CommentEditAction": undefined,
                  "clickDelete": undefined,
                  "clickDisLike": undefined,
                  "clickEdit": undefined,
                  "clickLike": undefined,
                  "clickReply": undefined,
                  "id": undefined,
                  "likes": undefined,
                  "onClick": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "view all...",
                  "className": "col-4 text-primary",
                  "onClick": undefined,
                  "onKeyUp": [Function],
                  "role": "button",
                  "tabIndex": "0",
                },
                "ref": null,
                "rendered": "view all...",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Connect(ReplyList)
                    CommentDeleteAction={[Function]}
                    parentId={1}
                    replyList={Array []}
                    slug="mock-slug1"
                  />,
                  "className": "d-none",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "CommentDeleteAction": [Function],
                    "history": undefined,
                    "onSubmitReply": undefined,
                    "parentId": 1,
                    "repliesShow": undefined,
                    "replyList": Array [],
                    "slug": "mock-slug1",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Connect(ReplyBox)
                    art_slug="mock-slug1"
                    childId={1}
                    isEdit={false}
                    parentId={1}
                  />,
                  "className": "d-none",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "art_slug": "mock-slug1",
                    "childId": 1,
                    "isEdit": false,
                    "parentId": 1,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
